src/charts/sparkline/sparkline-chart.directive.js

```diff
@@ -0,0 +1,327 @@
+                        '</div>';
+                }
+                else if (scope.config.tooltipType === 'usagePerDay') {
+                  var dayUsage = Math.round(d[0].value / scope.chartData.total * 100.0);
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <th colspan="2">' + d[0].x.toLocaleDateString() + '</th>' +
+                    '      </tr>' +
+                    '      <tr>' +
+                    '        <td class="name">' + dayUsage + '%:' + '</td>' +
+                    '        <td class="value text-nowrap">' + d[0].value + ' ' + scope.config.totalUnits + ' ' + d[0].name + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else if (scope.config.tooltipType === 'valuePerDay') {
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <td class="value">' +  d[0].x.toLocaleDateString() + '</td>' +
+                    '        <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else if (scope.config.tooltipType === 'percentage') {
+                  var percentUsed = Math.round(d[0].value / scope.chartData.total * 100.0);
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <td class="name">' + percentUsed + '%' + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else {
+                  tip = c3ChartDefaults.getDefaultSparklineTooltip().contents(d);
+                }
+                return tip;
+              },
+              position: function (data, width, height, element) {
+                try {
+                  var center = parseInt(element.getAttribute('x'));
+                  var top = parseInt(element.getAttribute('y'));
+                  var chartBox = document.querySelector('#' + scope.sparklineChartId).getBoundingClientRect();
+                  var graphOffsetX = document.querySelector('#' + scope.sparklineChartId + ' g.c3-axis-y').getBoundingClientRect().right;
+                  var x = Math.max(0, center + graphOffsetX - chartBox.left - Math.floor(width / 2));
```

Comment:
Yeah, I can see that.  It's probably more of a habitual thing for me since we do single line for `var`s in katello.  But I am fine with both styles as long as the `var`s are at the top.